configuration.go

```diff
@@ -32,6 +32,19 @@ func (s ServerSuffrage) String() string {
 	return "ServerSuffrage"
 }
 
+// ConfigurationStore provides an interface that can be implemented by FSMs
+// to store configuration updates made in the replicated log.
```

Comment:
Might be nice to add a line to say why they might want to as a hint for people trying to figure out how to implement raft themselves. Something like. "In general this is only necessary for FSMs that mutate durable state directly instead of applying changes in memory and snapshotting periodically. By storing configuration changes, the persistent FSM state can behave as a complete snapshot, and be able to recover without an external snapshot just for persisting the raft configuration."